adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,50 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <algorithm>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    std::stack<char> stack;
+
+    for (auto& c : source)
+    {
+        auto it = std::find_if(brackets.begin(), brackets.end(), [&](const std::pair<char, char>& p) -> bool {
```

Comment:
please try to decompose this function - create helper function like isBracket(),